Subject: Handle pre-3.0.14 providers

The get_params and gettable_params function pointers were only added
to the function table in 3.0.14.  We need to cope with providers which
do not provide them (e.g. the FIPS module).

Index: os-openssl/crypto/evp/pmeth_lib.c
===================================================================
--- os-openssl.orig/crypto/evp/pmeth_lib.c
+++ os-openssl/crypto/evp/pmeth_lib.c
@@ -1061,8 +1061,13 @@ static int evp_pkey_ctx_add1_octet_strin
     os_params[0] = OSSL_PARAM_construct_octet_string(param, NULL, 0);
     os_params[1] = OSSL_PARAM_construct_end();
 
-    if (!EVP_PKEY_CTX_get_params(ctx, os_params))
+    if (!EVP_PKEY_CTX_get_params(ctx, os_params)) {
+        if (EVP_PKEY_CTX_gettable_params(ctx) == NULL) {
+            /* Older provider that doesn't support gettable parameters */
+            return evp_pkey_ctx_set1_octet_string(ctx, fallback, param, op, ctrl, data, datalen);
+        }
         return 0;
+    }
 
     /* Older provider that doesn't support getting this parameter */
     if (os_params[0].return_size == OSSL_PARAM_UNMODIFIED)
